### PR TITLE
feat: implement pagination

### PR DESCRIPTION
**Pull request by orisano**
for #19

### CODE REVIEW

File: render.go [hunk 1]
```diff
@@ -72,24 +78,34 @@ func (r *Render) renderWindowTooSmall() {
...
 		suggestions,
-		int(r.col)-len(r.prefix),
+		int(r.col)-len(r.prefix)-1,
 	)
+	formatted = formatted[completions.verticalScroll : completions.verticalScroll+windowHeight]
 	l := len(formatted)
-	r.prepareArea(l)
+	r.prepareArea(windowHeight)
 
 	d := (len(r.prefix) + len(buf.Document().TextBeforeCursor())) % int(r.col)
```
Comment (c-bata) [inline render.go]: Please rewrite this line to following code.

```
// +1 means a width of scrollbar.
d := (len(r.prefix) + len(buf.Document().TextBeforeCursor()) + 1) % int(r.col)
```

File: render.go [hunk 2]
```diff
@@ -98,23 +114,32 @@ func (r *Render) renderCompletion(buf *Buffer, completions *CompletionManager) {
...
 		r.out.WriteStr(formatted[i].Description)
-		r.out.CursorBackward(width)
+
+		if isScrollThumb(i) {
+			r.out.SetColor(DefaultColor, r.scrollbarThumbColor, false)
+		} else {
+			r.out.SetColor(DefaultColor, r.scrollbarBGColor, false)
+		}
+		r.out.WriteStr(" ")
+		r.out.CursorBackward(width + 1)
```
Comment (c-bata) [inline render.go]: Please add comment.
```
// +1 means a width of scrollbar.
```

**Comment by c-bata** [comment]
Great job!! This is almost perfect.

**Comment by orisano** [comment]
I fixed it

**Comment by c-bata** [comment]
Awesome!

**Pull request merged**